include/crud-terms.php

```diff
@@ -293,54 +293,19 @@ public function set_pre_term_name( $name ) {
 	 * @return string Slug with a language suffix if found.
 	 */
 	public function set_pre_term_slug( $slug, $taxonomy ) {
-		if ( ! $this->model->is_translated_taxonomy( $taxonomy ) ) {
-			return $slug;
-		}
-
-		if ( ! $slug ) {
-			$slug = sanitize_title( $this->pre_term_name );
-		}
+		$term_slug = new PLL_Term_Slug( $this->model );
 
-		if ( ! term_exists( $slug, $taxonomy ) ) {
-			return $slug;
+		$term_data = $term_slug->get_term_data( $this->pre_term_name, $slug, $taxonomy );
+		if ( ! isset( $term_data['term_id'] ) ) {
+			// The slug must be returned un-suffixed.
+			return $term_data['slug'];
 		}
 
-		/**
-		 * Filters the subsequently inserted term language.
-		 *
-		 * @since 3.3
-		 *
-		 * @param PLL_Language|null $lang     Found language object, null otherwise.
-		 * @param string            $taxonomy Term taonomy.
-		 * @param string            $slug     Term slug
-		 */
-		$lang = apply_filters( 'pll_inserted_term_language', null, $taxonomy, $slug );
-
-		if ( ! $lang instanceof PLL_Language ) {
-			return $slug;
-		}
-
-		$parent = 0;
-		if ( is_taxonomy_hierarchical( $taxonomy ) ) {
-			/**
-			 * Filters the subsequently inserted term parent.
-			 *
-			 * @since 3.3
-			 *
-			 * @param int          $parent   Parent term ID, 0 if none.
-			 * @param string       $taxonomy Term taxonomy.
-			 * @param string       $slug     Term slug
-			 */
-			$parent = apply_filters( 'pll_inserted_term_parent', 0, $taxonomy, $slug );
-		}
-
-		$term_id = (int) $this->model->term_exists_by_slug( $slug, $lang, $taxonomy, $parent );
-
 		// If no term exist in the given language with that slug, it can be created.
-		if ( ! $term_id ) {
-			$slug .= '-' . $lang->slug;
+		if ( ! $term_data['term_id'] ) {
+			$term_data['slug'] .= '-' . $term_data['lang']->slug;
```

Comment:
Is it possible to pass the separator in parameter and return the whole slug from `PLL_Term_Slug` method?
So it avoid to return an array and allow to move these lines inside your new class?